src/meta_schedule/feature_extractor/utils.h

```diff
@@ -244,65 +246,101 @@ inline IntVec RelaxAndUnion(const std::vector<MultiIndex>& multi_indices, int64_
   }
   return access_shape;
 }
-
 /*!
- * \brief Given a list of multi-index pattern, return the minimal stride of a variable on it
- * \param multi_indices The list of multi-index pattern
- * \param buffer_stride The stride of the buffer
- * \param var The variable to be checked
- * \return The minimal stride of the variable on the multi-index pattern
+ * \brief Calculate the size of the regions unioned together.
+ * \param regions The regions that the buffer is accessed
+ * \param access_shape The shape of the data access
+ * \param analyzer The analyzer that contains the domain information
+ * \return The size of the single region after union
  */
-inline int64_t GetVarStride(const std::vector<MultiIndex>& multi_indices,
-                            const IntVec& buffer_stride, const Var& var) {
-  class CoefficientExtractor : private ExprVisitor {
-   public:
-    static int64_t Extract(const PrimExpr& expr, const Var& var) {
-      CoefficientExtractor extractor(var);
-      extractor.VisitExpr(expr);
-      return (extractor.visited_var && !extractor.visited_mul && !extractor.visited_add)
-                 ? 1
-                 : (extractor.visited_var ? extractor.stride : 0);
+inline int64_t RegionUnionSize(const std::vector<NDIntSet>& regions,
```

Comment:
The method `RegionUnionSize` serves the functionality very much alike `RelaxAndUnion`

```python
RelaxAndUnion(
  multi_indices: List[MultiIndex],   # e.g. [(i, j, k), (i + 1, j + 1, k + 1)]
  analyzer: arith.Analyzer,
) -> (
  numel: int64_t,
  access_shape: IntVec,
)

RegionUnionSize(
  regions: List[NDIntSet],  # e.g. [(i : i + 1, j : j + 1), (i : i + 2, j : j + 2)]
  analyzer: arith.Analyzer,
) -> (
  numel: int64_t,
  access_shape: IntVec,
)
```

Let's tweak the implementation a bit to make sure their function signature remains similar

Reply:
Addressed :)